cjdns-splice/src/lib.rs

```diff
@@ -187,6 +187,63 @@ pub fn re_encode<L: LabelT>(
     Ok(result)
 }
 
+// todo #3
+/// This will construct a label using an array representation of a path (`path_hops`), if any label along the `path_hops` needs to be re-encoded, it will be.
+pub fn build_label<L: LabelT>(path_hops: &[PathHop<L>]) -> Result<(L, Vec<L>)> {
+    if path_hops.len() < 2 {
+        return Err(Error::NotEnoughArguments);
+    }
+
+    if path_hops.first().unwrap().label_n.is_none() || path_hops.last().unwrap().label_p.is_none() {
+        return Err(Error::ZeroLabel);
+    }
+
+    if path_hops.first().unwrap().label_p.is_some() || path_hops.last().unwrap().label_n.is_some() {
+        return Err(Error::BadArgument);
+    }
+
+    let mut ret_path = vec![path_hops.first().unwrap().label_n.unwrap()];
```

Comment:
If the size is fixed and known in advance, it's a good practice to preallocate, e.g. `Vec::with_capacity`.